File: Data Structures & Algorithms/first-missing-positive/submission-6.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        n = len(nums)
        for i in range(n):
            nums[i] = max(nums[i], 0)
        for i in range(n):
            num = abs(nums[i])
            if 0 < num <=n:
                if nums[num - 1] > 0:
                    nums[num - 1] *= -1
                elif nums[num - 1] == 0:
                    nums[num - 1] = -1
        print(nums)
        for i in range(n):
            if nums[i] >= 0:
                return i + 1
        return n + 1

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-6.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_map<int, int> map;
        int res{};
        for (const auto& num : nums) {
            if (!map[num]) {
                map[num] = map[num - 1] + map[num + 1] + 1;
                map[num - map[num - 1]] = map[num];
                map[num + map[num + 1]] = map[num];
                res = max(res, map[num]);
            }

        }
        return res;
    }
};
